e2-studio-star-rx72n-firmware/libs/rx_hcsr04/src/rx_hcsr04.c

```
@@ -1741,14 +2164,30 @@ rx_err_t rx_hcsr04_measure(rx_hcsr04_t* handle, rx_hcsr04_result_t* result)
   if (err != k_rx_ok) {
+    /* Disarm ISR to prevent stale active=true if trigger fails after arming */
+    if (handle->echo_mode == k_hcsr04_echo_irq) {
+      (void)rx_hcsr04_isr_disarm((uint8_t)handle->echo_irq);
+    }
     return err;
   }
 
-  /* Measure echo pulse duration */
-  err = internal_measure_echo_pulse(handle, &result->echo_time_us);
+  /* Measure echo pulse duration (dispatch based on mode) */
+  if (handle->echo_mode == k_hcsr04_echo_irq) {
```

Clang (14.0.6)

[warning] if with identical then and else branches

(bugprone-branch-clone)